lightning-net-tokio/src/lib.rs

```diff
@@ -42,65 +42,96 @@ pub struct Connection {
 	id: u64,
 }
 impl Connection {
-	fn schedule_read(peer_manager: Arc<peer_handler::PeerManager<SocketDescriptor>>, us: Arc<Mutex<Self>>, reader: futures::stream::SplitStream<tokio_codec::Framed<TcpStream, tokio_codec::BytesCodec>>) {
+	fn schedule_read(
+		peer_manager: Arc<peer_handler::PeerManager<SocketDescriptor>>,
+		us: Arc<Mutex<Self>>,
+		reader: futures::stream::SplitStream<tokio_codec::Framed<TcpStream, tokio_codec::BytesCodec>>,
+	) {
 		let us_ref = us.clone();
 		let us_close_ref = us.clone();
 		let peer_manager_ref = peer_manager.clone();
-		tokio::spawn(reader.for_each(move |b| {
-			let pending_read = b.to_vec();
-			{
-				let mut lock = us_ref.lock().unwrap();
-				assert!(lock.pending_read.is_empty());
-				if lock.read_paused {
-					lock.pending_read = pending_read;
-					let (sender, blocker) = futures::sync::oneshot::channel();
-					lock.read_blocker = Some(sender);
-					return future::Either::A(blocker.then(|_| { Ok(()) }));
-				}
-			}
-			//TODO: There's a race where we don't meet the requirements of disconnect_socket if its
-			//called right here, after we release the us_ref lock in the scope above, but before we
-			//call read_event!
-			match peer_manager.read_event(&mut SocketDescriptor::new(us_ref.clone(), peer_manager.clone()), pending_read) {
-				Ok(pause_read) => {
-					if pause_read {
+		tokio::spawn(
+			reader
```

Comment:
This is super awkward. Anything we can do with the defaults so it doesn't make a mess like this?

Reply:
I think changing this default might help: https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#function-calls

Reply:
Sure, lets try that.

Reply:
Should be better with this new change!